lxd/storage/backend_lxd.go

```diff
@@ -7093,36 +7093,78 @@ func (b *lxdBackend) GenerateInstanceBackupConfig(inst instance.Instance, snapsh
 			}
 		}
 
-		for _, device := range inst.ExpandedDevices() {
-			if device["type"] != "disk" || instancetype.IsRootDiskDevice(device) {
-				continue
-			}
+		var customVols []*backupConfig.Volume
+		var customVolPools []*api.StoragePool
 
-			// Custom storage volume might be in a different pool, so load the pool.
-			pool, err := LoadByName(b.state, device["pool"])
-			if err != nil {
-				return nil, err
-			}
+		if volBackupConf != nil {
```

Comment:
This part triggered me later on when i saw you are passing an empty (but non-nil) list when generating an export index file.

My feeling on this now is that this function `GenerateInstanceBackupConfig` is trying to do too much and instead we should be breaking out the generation of custom volume config selection into a separate function, so that we can just this function `GenerateInstanceBackupConfig` take the proposed `volBackupConf` (but called `customVolConf` or similar to indicate its just for custom volumes) and then have this function append them to its root volume config.

Then we can have the call sites decide whether or not to include custom volume config, and it should then also be possible to be more efficient about loading that data too.

What do you think?